x/wasm/client/cli/query.go

```diff
@@ -168,3 +185,30 @@ func GetCmdGetContractState(cdc *codec.Codec) *cobra.Command {
 		},
 	}
 }
+func GetCmdGetContractStateRaw(cdc *codec.Codec) *cobra.Command {
+	return &cobra.Command{
+		Use:   "raw [bech32_address] [key]",
+		Short: "Prints out internal state for key of a contract given its address",
+		Long:  "Prints out internal state for of a contract given its address",
+		Args:  cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cliCtx := context.NewCLIContext().WithCodec(cdc)
+
+			addr, err := sdk.AccAddressFromBech32(args[0])
+			if err != nil {
+				return err
+			}
+			key := args[1]
+			if key == "" {
+				return errors.New("key must not be empty")
+			}
+			route := fmt.Sprintf("custom/%s/%s/%s/%s", types.QuerierRoute, keeper.QueryGetContractState, addr.String(), key)
```

Comment:
It would be good to include `all`/`raw`/`smart` here to make it easier to differentiate.
Adding one length gets tricky when we add a third.

Reply:
btw, smart should upload a json blob, which doesn't really fit in the path domain, you can use `cliCtx.QueryWithData` to include a data field (this request body in HTTP land), which is nicer place to dump json

Reply:
> The key can be anything

I would also move this to the body. Can either be raw or wrapped into some
```json
{
"key": "the data"
}
```
element to make it more restful. WDYT?

Reply:
For the smart query as well. Should it be json or length prefixed amino? I guess later one

Reply:
> element to make it more restful. WDYT?

Hmmm... I wasn't thinking of restful at all with the cli, but then yes, the Querier implementation will be used by both, so it makes sense for something that works well as a REST backend. Of course the rest-server can make any needed transform from the user input to the Querier interface, so the most important things are to:

1. Document the API and
2. Provide a helper function to construct the querier call

The internal api between rest/cli and QueryHandler in the abci app is not so important (if helper function is exposed), but yeah, should not be too ugly. Use you judgement there

Reply:
I would start with raw bytes. A REST server impl would also need to take the result types into account. I noticed an empty result is currently `nil` not `[]` for example.